crates/src/random/random.cairo

```diff
@@ -0,0 +1,98 @@
+//! Random struct and methods for generating random numbers with interval [min_number, max_number]
+
+// Core imports
+use poseidon::PoseidonTrait;
+use hash::HashStateTrait;
+use traits::Into;
+
+// Random struct for managing random number generation
+#[derive(Drop)]
+struct Random {
+    seed: felt252,
+    nonce: felt252,
+}
+
+// Random trait for generating random numbers with interval [min_number, max_number]
+trait RandomTrait {
+    /// Returns a new `Random` struct.
+    /// # Arguments
+    /// * `seed` - A seed to initialize the random number generator.
+    /// # Returns
+    /// * The initialized `Random`.
+    fn new(seed: felt252) -> Random;
+
+    /// Returns a random number within the specified interval.
+    /// # Arguments
+    /// * `min_number` - The lower bound of the interval.
+    /// * `max_number` - The upper bound of the interval.
+    /// # Returns
+    /// * A random number within the interval [min_number, max_number].
+
+    fn getRandomWithInterval(ref self: Random, min_number: u256, max_number: u256) -> u256;
```

Comment:
`getRandomWithInterval` should be in snake case.

Also `random.get_random...` is a bit ... you know, what about `random.get_within_interval`